internal/pkg/deploy/pipeline.go

```diff
@@ -607,14 +634,101 @@ func (stg *PipelineStage) Deployments() ([]DeployAction, error) {
 	return actions, nil
 }
 
-// PreDeployments returns a list of pre-deployment actions for the pipeline stage.
-func (stg *PipelineStage) PreDeployments() ([]Step, error) {
-	return nil, nil
+// PostDeployments returns a list of post-deployment actions for the pipeline stage.
+func (stg *PipelineStage) PostDeployments() ([]PrePostDeployAction, error) {
+	if len(stg.postDeployments) == 0 {
+		return nil, nil
+	}
+
+	var prevActions []orderedRunner
+	if approval := stg.Approval(); approval != nil {
+		prevActions = append(prevActions, approval)
+	}
+	preDeployActions, err := stg.PreDeployments()
+	if err != nil {
+		return nil, err
+	}
+	for i := range preDeployActions {
+		prevActions = append(prevActions, &preDeployActions[i])
+	}
```

Comment:
Why do we need this? I thought we did that in `stg.Deployments()`

Reply:
What if they don't specify any deployments but only pre- and post-deployment actions?

Reply:
I don't remember we decided on this but I feel like deployments should be the core and if there's no deployments pre/post deployments doesn't make a lot of sense. Im also open to other ideas.

Reply:
hmmm interesting - I am not aware of any validation that prevents `deployments` from being empty...

This opens a door for the kind of use case where people want to do some action as a separate stage. I can't think of a specific example though.

Reply:
And if people want pre- and post-deployment actions but don't want their workload deployments to run in parallel, this is how they would do it.